crates/mbe/src/expander/matcher.rs

```diff
@@ -729,17 +730,16 @@ fn match_meta_var(kind: &str, input: &mut TtIter) -> ExpandResult<Option<Fragmen
     input.expect_fragment(fragment).map(|it| it.map(Fragment::Tokens))
 }
 
-fn collect_vars(buf: &mut Vec<SmolStr>, pattern: &MetaTemplate) {
+fn collect_vars<F: FnMut(SmolStr)>(collector_fun: &mut F, pattern: &MetaTemplate) {
```

Comment:
no need to pass the closure by mut ref
bors d+

Reply:
rustc seems to disagree or maybe I miss something. I need a mut ref because the closure's call need a &mut self

Reply:
`mut collector_fun: F` should work
Calling the closure requires `&mut F`, if you pass ownership the name needs to be mutable for rust to be able to borrow mutably when calling it